source/lib.ts

```diff
@@ -76,18 +76,23 @@ const promisifyStore = (passedStore: LegacyStore | Store): Store => {
 	return new PromisifiedStore()
 }
 
+type OptionsWithLegacyStore = Omit<Partial<Options>, 'store'> & {
+	store?: Store | LegacyStore
+}
+
 /**
  * Type-checks and adds the defaults for options the user has not specified.
  *
  * @param options {Options} - The options the user specifies.
  *
  * @returns {Options} - A complete configuration object.
  */
-const parseOptions = (
-	passedOptions: Omit<Partial<Options>, 'store'> & {
-		store?: Store | LegacyStore
-	},
-): Options => {
+const parseOptions = (passedOptions: OptionsWithLegacyStore): Options => {
```

Comment:
If we add the `Configuration` interface, this function should return it:

```suggestion
const parseOptions = (passedOptions: OptionsWithLegacyStore): Configuration => {
```